src/components/InputOption.js

```diff
@@ -6,6 +6,25 @@ import {
 } from 'react-bootstrap'
 
 class InputOption extends Component {
+
+  handleChange = (event, key) => {
+    const option = this.props.options[this.props.optionKey]
+    const model = this.props.model
+    const optionType = option.type || 'number'
+    let attrs = {}
+    let value = event.target.value
+
+    if (optionType === 'number') {
+      value = value === '' ? '' : parseFloat(value)
+    }
+
+    attrs[key] = value
+    if (option.onChange !== undefined) {
+      attrs = option.onChange(attrs, model)
+    }
+    this.props.onChange(attrs)
```

Comment:
I think we could probably do this without a doubled state by setting a flag on attrs.